src/layers/legacy/medCoreLegacy/database/medAbstractDatabaseImporter.cpp

```diff
@@ -1000,13 +1005,39 @@ QString medAbstractDatabaseImporter::generateUniqueVolumeId ( const medAbstractD
     for( QString orient : orientations )
     {
         double d_orient = orient.toDouble();
-        orientation += QString::number ( d_orient, 'g', 4 );
+        orientation += QString::number ( d_orient, 'f', 4 );
     }
 
     // define a unique key string to identify which volume an image belongs to.
-    // we use: patientName, studyId, seriesId, orientation, seriesNumber, sequenceName, sliceThickness, rows, columns.
+    // we use: patientName, studyId, seriesId, orientation, seriesNumber, sequenceName, sliceThickness, rows, columns,
+    // echoTime and flipAngle.
     // All images of the same volume should share similar values of these parameters
-    QString key = patientName+studyDicomId+seriesDescription+orientation+sequenceName+sliceThickness+rows+columns;
+    QString key;
+    if(isDicom)
+    {
+        key= patientName+studyDicomId+seriesNumber+orientation+sequenceName+sliceThickness+rows+columns
+            +echotime+flipAngle;
+    }
+    else
+    {
+        key= patientName+studyDicomId+seriesDescription+orientation+sequenceName+sliceThickness+rows+columns;
+    }
 
     return key;
 }
+
+//-----------------------------------------------------------------------------------------------------------
+/**
+* Look for dicom extension in file name
+* @param fileName - name of the file that we try to import
+* @return  true if dicom file, false otherwise
+**/
+bool medAbstractDatabaseImporter::isDicomName(const QString & fileName)
```

Comment:
Please replace by
```C++
bool medAbstractDatabaseImporter::isDicomName(const QString & fileName)
{
    QFileInfo info(fileName);
    return info.suffix().toLower() == "dcm";
}

Reply:
Done and good catch I do not know why i didn't do it like that from the start

Reply:
Thanks